tests/components/airly/test_init.py

```diff
@@ -88,37 +90,77 @@ async def test_config_with_turned_off_station(hass, aioclient_mock):
 
 async def test_update_interval(hass, aioclient_mock):
     """Test correct update interval when the number of configured instances changes."""
-    entry = await init_integration(hass, aioclient_mock)
-
-    assert len(hass.config_entries.async_entries(DOMAIN)) == 1
-    assert entry.state == ENTRY_STATE_LOADED
-    for instance in hass.data[DOMAIN].values():
-        assert instance.update_interval == timedelta(minutes=15)
-
-    entry = MockConfigEntry(
-        domain=DOMAIN,
-        title="Work",
-        unique_id="66.66-111.11",
-        data={
-            "api_key": "foo",
-            "latitude": 66.66,
-            "longitude": 111.11,
-            "name": "Work",
-        },
-    )
-
-    aioclient_mock.get(
-        "https://airapi.airly.eu/v2/measurements/point?lat=66.660000&lng=111.110000",
-        text=load_fixture("airly_valid_station.json"),
+    headers = {"X-RateLimit-Limit-day": "100", "X-RateLimit-Remaining-day": "15"}
+    point = datetime(
+        year=2020, month=3, day=2, hour=21, minute=20, second=0, tzinfo=NATIVE_UTC
     )
-    entry.add_to_hass(hass)
-    await hass.config_entries.async_setup(entry.entry_id)
-    await hass.async_block_till_done()
 
-    assert len(hass.config_entries.async_entries(DOMAIN)) == 2
-    assert entry.state == ENTRY_STATE_LOADED
-    for instance in hass.data[DOMAIN].values():
-        assert instance.update_interval == timedelta(minutes=30)
+    with patch("homeassistant.core.dt_util.utcnow") as mock_utcnow:
+        mock_utcnow.return_value = point
+
+        entry = MockConfigEntry(
+            domain=DOMAIN,
+            title="Home",
+            unique_id="123-456",
+            data={
+                "api_key": "foo",
+                "latitude": 123,
+                "longitude": 456,
+                "name": "Home",
+            },
+        )
+
+        aioclient_mock.get(
+            API_POINT_URL,
+            text=load_fixture("airly_valid_station.json"),
+            headers=headers,
+        )
+        entry.add_to_hass(hass)
+        await hass.config_entries.async_setup(entry.entry_id)
+        await hass.async_block_till_done()
+
+        assert aioclient_mock.call_count == 1
+        assert len(hass.config_entries.async_entries(DOMAIN)) == 1
+        assert entry.state == ENTRY_STATE_LOADED
+        await hass.async_block_till_done()
+
+        instance1 = list(hass.data[DOMAIN].values())[0]
```

Comment:
Don't access integration details in the test. Move time forward instead and check that updates were made or not according to our expectations.

https://github.com/home-assistant/core/blob/c80f34a75454610bc337f6b22790e5bf52542308/tests/common.py#L363-L365

Reply:
I'm not sure if I did it correctly because the test only passes for a future date. What have I missed?